src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/StringCollectionTextWriter.cs

```diff
@@ -62,7 +70,19 @@ public override void Write([NotNull] char[] buffer, int index, int count)
                 throw new ArgumentOutOfRangeException(nameof(count));
             }
 
-            Content.Append(buffer, index, count);
+            while (count > 0)
+            {
+                // Split large char arrays into 1KB strings.
+                var currentCount = count;
+                if (MaxCharToStringLength < currentCount)
+                {
+                    currentCount = MaxCharToStringLength;
+                }
+
+                _content.Append(new string(buffer, index, currentCount));
+                index += currentCount;
+                count -= currentCount;
+            }
```

Comment:
Perhaps the `WriteTo(char[], int, int)` method could be removed from `BufferedHtmlContent`. I don't see other uses.

Even then `StringCollectionTextWriterContent` should be cleaner as a `BufferedHtmlContent` subclass. Why did you choose to duplicate?

Separately, might be worth explaining at the top of this block _why_ limiting ourselves to 1k blocks is important.